src/descriptor/mod.rs

```diff
@@ -261,14 +261,15 @@ pub enum DescriptorType {
 }
 
 impl DescriptorType {
-    /// Whether this is a segwit descriptor.
+    /// Returns the segwit version implied by the descriptor type.
     ///
-    /// Returns true whether it is "native" segwit or "wrapped" p2sh segwit
-    pub fn is_segwit(&self) -> bool {
+    /// This will return `Some(0)` whether it is "native" segwitv0 or "wrapped" p2sh segwit.
+    pub fn segwit_version(&self) -> Option<u8> {
```

Comment:
nit: There is a `[WitnessVersion](https://docs.rs/bitcoin/0.28.0/bitcoin/util/address/enum.WitnessVersion.html)` struct upstream that we can use here. 

You need not fix this in the current PR as it is good to go from my end, I can open it up as a good first issue.

Reply:
I added a fixup commit for this.